ethcore/light/src/on_demand/mod.rs

```diff
@@ -180,6 +185,14 @@ impl Pending {
 		self.net_requests = builder.build();
 		self.required_capabilities = capabilities;
 	}
+
+	// return no response, will result in an error
+	// consume object (similar to drop when no reply found)
```

Comment:
Best comment (object is definitely a bad wording for rust) : 
// returning no reponse, it will result in an error.
// self is consumed on purpose.
The comment is just here to point out that we do not want self to be use back (used oneshot channel).

Reply:
Yep, I think the comment is useful here. I like your suggestion for an update.